lib/utils/expando.js

```diff
@@ -0,0 +1,221 @@
+import { $ } from '../vendor';
+import expandoTemplate from '../templates/expando.mustache';
+
+// @type {WeakMap.<expandoButton, Expando>}
+export const expandos = new WeakMap();
+
+// Used for specifying primary expando for any given link
+// @type {Object.<string, Expando>} - key: href
+export const primaryExpandos = {};
```

Comment:
I somewhat want this to be a `Map` for symmetry with `expandos`, but I don't feel strongly about that. Feel free to leave it as-is.